contractcourt/channel_arbitrator.go

```diff
@@ -850,6 +856,25 @@ func (c *ChannelArbitrator) stateStep(
 		case chainTrigger, userTrigger:
 			log.Infof("ChannelArbitrator(%v): noop trigger %v",
 				c.cfg.ChanPoint, trigger)
+
+			// The commitment transaction has been broadcast, but it
+			// doesn't necessarily need to be the commitment
+			// transaction version that is going to be confirmed. To
+			// be sure that any of those versions can be anchored
+			// down, we now submit all anchor resolutions to the
+			// sweeper. The sweeper will keep trying to sweep all of
+			// them.
+			anchors, err := c.cfg.Channel.NewAnchorResolutions()
```

Comment:
This state will potentially be executed several times. Need to ensure this operation is idempotent, or we could add new state for anchor sweep publication (could also possibly be easier to test).

Reply:
Afaik there is no side-effect of this call. And I also believe it returns the same resolutions every time. Do you see a problem somewhere?

Reply:
Comment added about idempotency